### PR TITLE
Add rocket

### PR DESCRIPTION
**Pull request by ThomasRochais**
### What changed
- Add rocket class
- Changed the display text after game starts to black window with the rocket

### Manual test
1. The menu is still working
2. When entering the "Start Game" sub menu a rocket is displayed over a black background
3. The rocket can be moved within the boundaries of the screen

### Potential problems
- To keep the rocket moving when holding a key I added the line
pygame.key.set_repeat(10,5) in game.py 
however this affect the navigation in the menu

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
+import pygame
```
Comment (Hamtara) [inline main.py:1]: Can you remove this import and the following that's commented.

File: rocket.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import pygame
+import os 
+from menu import *
+#First let's make a moving rectangle here and then modify to make a moving rocket
+class Rocket():
+    def __init__(self, game):
+        self.game = game #Use a game object
+        self.width, self.height = self.game.DISPLAY_W / 20, self.game.DISPLAY_H / 20#Size of the rectangle
```
Comment (Hamtara) [inline rocket.py:8]: nit: try to keep the aspect ratio

```suggestion
        self.width, self.height = self.game.DISPLAY_W / 20, self.game.DISPLAY_H / 20# Size of the rectangle
```

File: rocket.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        #Initial position of the rectangle (bottom, centered)
+        self.x, self.y = self.game.DISPLAY_W / 2 - self.width/2, self.game.DISPLAY_H - self.height
+        self.vel=1 #Moving velocity
+        self.spaceship = pygame.transform.scale(pygame.image.load(os.path.join("assets/sprites", "spaceship-a.svg")), (round(self.width),round(self.height)))
+
+
+    def blit_rocket(self):
+        self.game.display.blit(self.spaceship, (self.x,self.y))
+        #pygame.draw.rect(self.game.display,(255,0,0),(self.x,self.y,self.width,self.height))
```
Comment (Hamtara) [inline rocket.py:17]: Is this meant to stay?

File: rocket.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        if self.x + self.vel < self.game.DISPLAY_W - self.width: 
+            self.x += self.vel
+
+    def move_up(self):
+        if self.y - self.vel > 0:
+            self.y -= self.vel
+
+    def move_down(self):
+        if self.y + self.vel < self.game.DISPLAY_H - self.height:
+            self.y += self.vel
```
Comment (Hamtara) [inline rocket.py:33]: Adding a line at the end of the file would be great.
```suggestion
            self.y += self.vel
            
```

File: game.py [hunk 5]
```diff
@@ -43,13 +51,22 @@ def check_events(self):
...
 
     def draw_text(self, text, size, x, y ):
         font = pygame.font.Font(self.font_name,size)
         text_surface = font.render(text, True, self.WHITE)
         text_rect = text_surface.get_rect()
         text_rect.center = (x,y)
         self.display.blit(text_surface,text_rect)
+
+    def redrawGameWindow(self):
+        self.rocket.blit_rocket()#Draw the rocket
```
Comment (Hamtara) [inline game.py:70]:
```suggestion
        self.rocket.blit_rocket() #Draw the rocket
```

File: game.py [hunk 6]
```diff
@@ -43,13 +51,22 @@ def check_events(self):
...
+        self.window.blit(self.display, (0,0))#Blitting is drawing
```
Comment (Hamtara) [inline game.py:71]:
```suggestion
        self.window.blit(self.display, (0,0)) #Blitting is drawing
```

File: game.py [hunk 7]
```diff
@@ -43,13 +51,22 @@ def check_events(self):
...
                     self.UP_KEY = True
-
+                if event.key == pygame.K_LEFT:
+                    self.LEFT_KEY = True
+                if event.key == pygame.K_RIGHT:
+                    self.RIGHT_KEY = True
+                    
     def reset_keys(self):
-        self.UP_KEY, self.DOWN_KEY, self.START_KEY, self.BACK_KEY = False, False, False, False
+        self.LEFT_KEY, self.RIGHT_KEY,self.UP_KEY, self.DOWN_KEY, self.START_KEY, self.BACK_KEY = False, False, False, False, False, False
```
Comment (Hamtara) [inline game.py:60]: The line is becoming quite long we might want to consider splitting it at least in two 
1. Right and Left
2. Up and Down

File: game.py [hunk 8]
```diff
@@ -15,20 +16,27 @@ def __init__(self):
         self.options = OptionsMenu(self)
         self.credits = CreditsMenu(self)
         self.curr_menu = self.main_menu
+        self.rocket=Rocket(self)
 
     def game_loop(self):
         while self.playing:
+            #self.draw_text('Thanks for Playing', 20, self.DISPLAY_W/2, self.DISPLAY_H/2)
```
Comment (Hamtara) [inline game.py:23]: We can remote it at this time I think. Until we do a game over screen.

File: game.py [hunk 9]
```diff
@@ -1,11 +1,12 @@
...
+from pygame import key
 from menu import *
-
+from rocket import *
 class Game():
     def __init__(self):
         pygame.init()
         self.running, self.playing = True, False
-        self.UP_KEY, self.DOWN_KEY, self.START_KEY, self.BACK_KEY = False, False, False, False
+        self.LEFT_KEY, self.RIGHT_KEY, self.UP_KEY, self.DOWN_KEY, self.START_KEY, self.BACK_KEY = False, False, False, False, False, False
```
Comment (Hamtara) [inline game.py:9]: Same comment this line is quite long.

File: rocket.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
+import pygame
+import os 
+from menu import *
```
Comment (Hamtara) [inline rocket.py:3]: This import doesn't seems to be needed.

File: main.py (hunk 1 above)
Comment (ThomasRochais) [reply main.py:1]: OK

File: rocket.py (hunk 3 above)
Comment (ThomasRochais) [reply rocket.py:17]: the commented out rectangle no
the function though, yeah. I thought it was cleaner that way

File: rocket.py (hunk 4 above)
Comment (ThomasRochais) [reply rocket.py:33]: OK, I guess why not

File: game.py (hunk 7 above)
Comment (ThomasRochais) [reply game.py:60]: OK, I'm splitting into 3 lines

File: rocket.py (hunk 2 above)
Comment (ThomasRochais) [reply rocket.py:8]: yes, I fixed it.

File: game.py [hunk 11]
```diff
@@ -1,40 +1,44 @@
...
         self.curr_menu = self.main_menu
-        self.rocket=Rocket(self)
+        self.rocket = Rocket(self)
 
     def game_loop(self):
         while self.playing:
-            #self.draw_text('Thanks for Playing', 20, self.DISPLAY_W/2, self.DISPLAY_H/2)
-            pygame.key.set_repeat(10,5)#So rocket can keep moving when pressing and holding keys
-            self.display.fill(self.BLACK)#Black screen
+            pygame.key.set_repeat(
```
Comment (Hamtara) [inline game.py:27]: Putting the comment on top would prevent the weird line break.

File: .vscode/settings.json [hunk 12]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (Hamtara) [inline .vscode/settings.json:1]: Can you remove it from the commit and add to the .gitignore at the end .vscode/ 🙏

**Review by Hamtara — COMMENTED** [review]
Otherwise good work. 🎉

File: .vscode/settings.json (hunk 12 above)
Comment (ThomasRochais) [reply .vscode/settings.json:1]: I think I did it...

**Review by Hamtara — APPROVED** [review]

**Pull request merged**